chore: Add use cases to developer guide

CloudHill opened the pull request:
Added the following use cases to the developer guide:

* `UC C1` List Customers
* `UC C2` View Customer
* `UC C3` Delete Customer
* `UC C4` Update Customer
* `UC C5` Bookmark or Unbookmark Customer
* `UC C6` Assign Individual to Enterprise
* `UC C7` Create Customer Note
* `UC C8` Append Customer Note ➕
* `UC C9` Delete Customer Note
* `UC O1` List Orders
* `UC O2` Create Order ➕
* `UC O3` View Order
* `UC O4` Delete Order
* `UC O5` Update Order
* `UC O6` Advance Order Status
* ~~`UC O6` Tag Order to Customer~~
* `UC R1` List Rewards
* `UC R2` View Reward
* `UC R3` Delete Reward
* `UC R4` Update Reward
* `UC R5` Set/Add Customer Points ➕
* `UC R6` Give Reward to Customer

Closes #14

CloudHill committed:
chore: Add use cases to developer guide

JavonTeo commented on docs/DeveloperGuide.md:
```diff
@@ -283,32 +283,391 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests list of customers
```
A full stop could be used to finish the sentence.

JavonTeo reviewed (changes requested):
LGTM. Just a nit to pick: following the use case example in the CS2103T textbook, every line had a full stop, though full stops were not used much here. Other than that, everything looks good!

CloudHill committed:
chore: Update use cases

+ UC C8
+ UC O2
- UC O6
+ UC R5

CloudHill commented:
Added the full stops, also updated the list of use cases:

* `+ UC C8` Append Customer Note
* `+ UC O2` Create Order
* `- UC O6` Tag Order to Customer
* `+ UC R5` Set/Add Customer Points

JavonTeo reviewed (approved):
LGTM
